xbmc/utils/EGLImage.cpp

```diff
@@ -61,7 +111,7 @@ CEGLImage::CEGLImage(EGLDisplay display) :
 
 bool CEGLImage::CreateImage(EglAttrs imageAttrs)
 {
-  CEGLAttributes<22> attribs;
+  CEGLAttributesVec attribs;
```

Comment:
Why change to heap allocation?

Reply:
So I could get the exact size of the attributes and not print garbage if the attributes weren't filled up. We need 22 attributes but often only 12 are used.

Reply:
You could add `Size()` to `CEGLAttributes` as well :-)

Reply:
I imagine I would have to parse the attributes then to find the end. Also, I think Size in that case should return the actual array size (which would be 22). I'm not really sure what is the better solution.

Reply:
Haven't tried, but can't you just take `m_writePosition`?